proposals/0249-commission-update-rules.md

```diff
@@ -0,0 +1,135 @@
+1` when calculating validator commissions. This is the same vote account state
+used to calculate the leader schedule for epoch `E` so it must already be
+available in-protocol. The only exception is inflation reward distribution for
+epoch `E == 0`. In that case, use the commission rate set by vote accounts in
+the genesis config.
+
+#### Commission Calculation
+
+As before, when calculating inflation reward commission, avoid overflow by first
+converting the pending inflation reward to a `u128` integer. Then, since commission
+rates are now denominated as basis points, multiply the reward by the minimum of
+the vote account's inflation reward commission rate and the maximum commission rate
+of `10,000` basis points. Use integer division to divide the result by `10,000` and
+disregard the remainder.
+
+Similarly, to calculate the delegated stake inflation reward portion, avoid
+overflow by first converting the pending inflation reward to a `u128` integer.
+Then, multiply the reward by the maximum of `0` and `10,000` subtracted by the
+vote account's inflation reward commission rate. Use integer division to divide
+the result by `10,000` and disregard the remainder.
+
+### Vote Program
+
+```rust
+pub enum VoteInstruction {
+    // ..
+    UpdateCommission {..} // 5u32
+    /// # Account references
+    ///   0. `[WRITE]` Vote account to be updated with the new commission
+    ///   1. `[SIGNER]` Withdraw authority
+    UpdateCommissionBps { // 17u32
+        commission_bps: u16,
+        kind: CommissionKind,
+    },
+}
+
+#[repr(u8)]
+pub enum CommissionKind {
+    InflationRewards = 0,
+    BlockRevenue = 1,
+}
+```
+
+#### UpdateCommission
+
+Update the core vote program to no longer restrict commission increases from
+happening during the first half of the epoch.
+
+#### UpdateCommissionBps
+
+A new instruction for setting a commission rate denominated in basis points will
+be added to the vote program with the enum discriminant value of `17u32` little
+endian encoded in the first 4 bytes.
+
+Perform the following checks:
+
+- If the number of account inputs is less than 1, return
+`InstructionError::NotEnoughAccountKeys`
+- If the vote account (index `0`) fails to deserialize, return
+`InstructionError::InvalidAccountData`
+- If the vote account's authorized withdrawer is not an account input for the
+instruction or is not a signer, return
+`InstructionError::MissingRequiredSignature`
+
+Update the corresponding field for the specified commission kind:
+
+- `CommissionKind::InflationRewards`: update the
+`inflation_rewards_commission_bps` field
+- `CommissionKind::BlockRevenue`: return
+`InstructionError::InvalidInstructionData`
+
+Note that the commission rate is allowed to be set to any `u16` value
+
+## Alternatives Considered
+
+NA
+
+## Impact
+
+- Validators will need to wait at least one full epoch before their commission
+updates are applied.
+
+- Validators will be able to set inflation rewards commission rates in basis
+points. 
```

Comment:
This presupposes that VoteStateV4 already exists, correct? If so, can you add a reference to that SIMD?

Reply:
I added a "Dependencies" section in https://github.com/solana-foundation/solana-improvement-documents/pull/249/commits/11759042a51adb5b01e2947f168a415ddc9bddb8